Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -6916,14 +6916,40 @@ def get_security_profiles_command(security_profile: str = None):
 
 
 @logger
-def apply_security_profile(xpath: str, profile_name: str) -> Dict:
+def apply_security_profile(xpath: str, profile_name: str, profile_type: str) -> Dict:
+    # first api call just for getting the current status
+    params = {
+        'action': 'get',
+        'type': 'config',
+        'xpath': xpath,
+        'key': API_KEY,
+    }
+    result = http_request(URL, 'GET', params=params)
+
+    # Get all profile types already existing, so we don't override them when updating
+    profile_types_result =\
+        result.get('response', {}).get('result', {}).get('entry', {}).get('profile-setting', {}).get('profiles', {})
+
+    # remove from the types the given profile type, since we update it anyway
+    profile_types = {'data-filtering', 'file-blocking', 'spyware', 'url-filtering',
+                     'virus', 'vulnerability', 'wildfire-analysis'} - {profile_type}
+
+    # first we update the given profile type with the given profile name
+    element = f"<{profile_type}><member>{profile_name}</member></{profile_type}>"
+
+    # Keeping the existing profile types
+    for p_type in profile_types:
+        if p_type in profile_types_result:
+            element += f"<{p_type}><member>{profile_types_result[p_type]['member']['#text']}</member></{p_type}>"
```

Comment:
`['#text']` is only true when the profile is un-committed, it fails if the rule was already committed.

use the `parse_pan_os_response` function to parse un-committed pan-os response to be like committed response

make sure to use `get` of `dict` instead also.